scripts/SetupServiceFabric/SetupServiceFabric.sh

```diff
@@ -26,6 +31,122 @@ if [[ "xenial" != "$Distribution" && "bionic" != "$Distribution" ]]; then
     exit -1
 fi
 
+# Check ServiceFabricRuntimePath and ServiceFabricSdkPath have been provided. If yes, extract that
+ServiceFabricRuntimePath=""
+ServiceFabricSdkPath=""
+for i in "$@"
+do
+case $i in
+    -sfrt=*|--servicefabricruntime=*)
+    ServiceFabricRuntimePath="${i#*=}"
+    ;;
+    -sfs=*|--servicefabricsdk=*)
+    ServiceFabricSdkPath="${i#*=}"
+    ;;
+    *)
+    echo Error: Unknown option passed in: $i
+    exit 1
+    ;;
+esac
+done
+
+# Check if systemd is running as PID1, if not it should be enabled via system-genie
+pidone=$(ps --no-headers -o comm 1)
+if [ "systemd" != "$pidone" ]; then
+    # Set system-genie to run systemd as PID 1
+    echo "Setting up system-genie to run systemd as PID 1"
+    echo "Installing .NET SDK and runtime 5.0"
+    wget https://packages.microsoft.com/config/ubuntu/$(lsb_release -rs)/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+    dpkg -i packages-microsoft-prod.deb
+    rm packages-microsoft-prod.deb
+
+    echo "Installing the .NET SDK"
+    apt-get update; \
+    apt-get install -y apt-transport-https && \
+    apt-get update && \
+    apt-get install -y dotnet-sdk-5.0
+
+    echo "Installing the .NET runtime "
+    apt-get update; \
+    apt-get install -y apt-transport-https && \
+    apt-get update && \
+    apt-get install -y aspnetcore-runtime-5.0
+
+    echo "Adding the wsl-translinux repository"
+    apt install apt-transport-https
+
+    wget -O /etc/apt/trusted.gpg.d/wsl-transdebian.gpg https://arkane-systems.github.io/wsl-transdebian/apt/wsl-transdebian.gpg
+    chmod a+r /etc/apt/trusted.gpg.d/wsl-transdebian.gpg
```

Comment:
can you avoid duplication of these steps from what you're doing and what is already happening ?

Reply:
Done.